### PR TITLE
Update OpenJ9 build for Windows

### PR DESCRIPTION
**Pull request by vsebe**
Update configuration for windows platform, switch bootjdk installation
directory to default location (home directory) for Windows machines.
Add optional error handling for intermittent failures.

Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

**Comment by vsebe** [comment]
@AdamBrousseau  please review, thank you!.

File: buildenv/jenkins/common/build.groovy [hunk 1]
```diff
@@ -720,4 +731,49 @@ def build_all() {
...
+    def maxRetry = 3
+    def retryCounter = 0
+    def doRetry = true
+
+    while ((maxRetry > retryCounter) && doRetry) {
+        retryCounter++
+        println("Attempt to recompile, retry: ${retryCounter}")
+
+        try {
+            sh "make clean && make ${EXTRA_MAKE_OPTIONS} ${make_target}"
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build.groovy:762]: Would we be smarter moving just the make $make_target command to it's own function so we aren't duplicating that code? I know it doesn't change often but if it does we have to change it in 2 spots.

**Comment by AdamBrousseau** [comment]
Jenkins compile win jdk8,jdk11,jdk17

**Comment by AdamBrousseau** [comment]
Jenkins compile win jdk8,jdk11,jdk17

**Comment by AdamBrousseau** [comment]
Jenkins compile win jdk8,jdk11,jdk17

**Review by AdamBrousseau — COMMENTED** [review]
This is good with me now. I would like a sign-off from @pshipton as well though. Once we let it soak a few days we can roll out more machines and delete the old ones, also pending Pete's approval.

**Review by AdamBrousseau — APPROVED** [review]
bump @pshipton for sign-off

File: buildenv/jenkins/common/build.groovy [hunk 2]
```diff
@@ -720,4 +734,49 @@ def build_all() {
...
+        if (matcher.find()) {
+            return true
+        }
+    }
+
+    println("Fail pattern not found!")
+    return false
+}
+
+def recompile(startTime) {
```
Comment (pshipton) [inline buildenv/jenkins/common/build.groovy:755]: Did we want to do something with startTime, such as print it out?

Comment (vsebe) [reply buildenv/jenkins/common/build.groovy]: removed

File: buildenv/jenkins/common/build.groovy [hunk 3]
```diff
@@ -720,4 +734,49 @@ def build_all() {
...
+    def maxRetry = 3
+    def retryCounter = 0
+    def doRetry = true
+
+    while ((maxRetry > retryCounter) && doRetry) {
+        retryCounter++
+        println("Attempt to recompile, retry: ${retryCounter}")
+
+        try {
+_            sh "make clean && ${get_compile_command()}"
```
[the remark this answers is not in the document]
Comment (vsebe) [reply buildenv/jenkins/common/build.groovy]: Good catch, thank you!
Removed.

**Comment by pshipton** [comment]
jenkins compile win jdk8

**Pull request merged**